ipaserver/install/ipa_custodia_regen.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (C) 2017  IPA Project Contributors, see COPYING for license
+"""Regenerate Custodia server.keys and config
+
+KEEP THIS SCRIPT COMPATIBLE WITH FREEIPA 4.5
+
+Christian Heimes <cheimes@redhat.com>
+"""
+from __future__ import print_function
+
+import logging
+import os
+
+from ipalib import api
+from ipaplatform.paths import paths
+from ipapython import admintool
+from ipapython.ipautil import backup_file
+from ipapython.ipa_log_manager import standard_logging_setup
+from ipaserver.install import installutils
+from ipaserver.install import sysupgrade
+from ipaserver.install.custodiainstance import CustodiaInstance
+
+
+logger = logging.getLogger(__name__)
+
+
+class IPACustodiaRegen(admintool.AdminTool):
+    command_name = 'ipa-custodia-regen'
+
+    usage = "%prog [options]\n"
+
+    @classmethod
+    def add_options(cls, parser):
+        super(IPACustodiaRegen, cls).add_options(parser)
+        parser.add_option(
+            "-r", '--regenerate', action="store_true",
+            dest="regenerate", default=False,
+            help="Perform regeneration"
+        )
+
+    def validate_options(self):
+        super(IPACustodiaRegen, self).validate_options(needs_root=True)
+        try:
+            installutils.check_server_configuration()
+        except RuntimeError as e:
+            raise admintool.ScriptError(e)
+        if not self.options.regenerate:
+            raise admintool.ScriptError(
+                "Nothing to do, --regenerate option not given."
+            )
+
+    def run(self):
+        super(IPACustodiaRegen, self).run()
+
+        # use private in-memory ccache
+        os.environ['KRB5CCNAME'] = 'MEMORY:ipa-custodia-regen'
+        # use host keytab to acquire TGT for LDAP connection
+        os.environ['KRB5_CLIENT_KTNAME'] = paths.KRB5_KEYTAB
+
+        if not api.isdone('finalize'):
+            api.bootstrap(
+                in_server=True,
+                debug=self.options.verbose,
+                confdir=paths.ETC_IPA,
+            )
+            api.finalize()
+        if not api.Backend.ldap2.isconnected():
+            api.Backend.ldap2.connect()
+
+        self.regen()
```

Comment:
maybe run a "consistency check" when --regenerate is not passed, that will tell you if running again with --regenerate is needed at all ?